parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -276,6 +276,10 @@ if [[ $OS == $UBUNTU_OS_NAME ]]; then
 fi
 {{end}}
 
+{{- if RunUnattendedUpgrades}}
+apt_get_update && apt_get_dist_upgrade && unattended_upgrade
```

Comment:
The big difference between unattended-upgrades and apt-get dist-upgrade is the list of things it will install.

Unattended upgrades is constrained to the list of updates that are deemed safe and vital for security/reliability.  They are not minor feature updates unless that was required for security.  (This is the default and recommended configuration for unattended-upgrade)

For example, on a test VM, I just logged in and noticed this right now:
```
58 packages can be updated.
4 updates are security updates.
```

After running unattended-upgrades on that machine (which normally cron does for me on regular basis), the login looks like this:
```
54 packages can be updated.
0 updates are security updates.    
```

This is very different from a full apt-get update/apt-get upgrade (which itself is less than apt-get dist-upgrade)

The actual ubuntu unattended-upgrade command will return an error if it fails to complete an update.  But it is constrained to the security updates. 

Another good thing about unattended-upgrades is that it does set the unattended settings for apt/apt-get/dpkg such that it should not hang (albeit, packages can still cause this problems but that is rare in the security patches).

Which to use is really a question of risks.  Balancing all of them.

We run unattended-upgrade on a regular basis because we can trust it at scale.

Reply:
PS - It is redundant to run unattended-upgrade after having done the full upgrade or dist-upgrade.

It may be useful to do unattended-upgrade first just to be sure they complete before getting into the larger set (both from a security standpoint and an ability to complete them)

So I would not run unattended afterwards.

Reply:
This all makes sense. What's perplexing is that, in practice, simply adding a "wait for apt locks and then run unattended-upgrade" during CSE *does not* in my tests produce the expected /var/run/reboot-required (a symptom of critical security updates arriving) outcome.

I'm going to try `apt-get update` && `unattended-upgrade` next.